src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -218,7 +218,7 @@ public boolean hasReminder(Reminder reminder) {
 
     @Override
     public void deletePerson(Person target) {
-        addressBook.removePerson(target);
+        contactTagIntegrationManager.deletePerson(target);
```

Comment:
I don't really think contactTagIntegrationManager should handle the addressBook.removePerson aspect as well. It feels to me that you should separate it into two commands. Address book still handles removing of the person. But you can remove the tag first before address book removes the person. Correct me if I am wrong. The rest of this PR looks great though!

Reply:
Well, I would argue that `ModelManager` is a Facade class. Having two methods implies that `ModelManager` is now handling logic that should be implemented elsewhere. 

Also, as mentioned in the PR, `deletePerson` does in fact affect `Tag`s as well. It could possibly remove a `Person`  that has `Tag1`. And `Tag1` might have only one contact which is this `Person`. This causes `Tag1` to be deleted. If `Tag2` is a parent-tag that only has `Tag1` as its only child tag and no direct contacts, then `Tag2` should also be deleted. Thus, I believe this, and the other edited methods should be implemented in `ContactTagIntegrationManager`. Given a choice, I would have refactored the entire `AddressBook` and `TagTree` into `ContactTagIntegrationManager` but it seems a little too late to do that. 

This was the cause of a bug that I found that I believe is the cause of the issue attached to this PR.

Reply:
I get your point, we will just do this then, I will approve it as it is. Thanks for the clarification.

Reply:
Thanks for pointing it out anyway. Appreciate the review!